modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -254,6 +262,17 @@ public void processOpts() {
         }
         additionalProperties.put(PLAY_VERSION, playVersion);
 
+        // Microprofile framework
+        if (additionalProperties.containsKey(MICROPROFILE_FRAMEWORK)) {
+            this.setMicroprofileFramework(additionalProperties.get(MICROPROFILE_FRAMEWORK).toString());
+
+            if (!MICROPROFILE_KUMULUZEE.equals(microprofileFramework)) {
+                throw new RuntimeException("Ivalid microprofileFramework '{}'. Must be 'kumuluzee' or none.");
```

Comment:
I'll fix the typo (Ivalid => Invalid) and string interpolation '{}' in another PR.